src/corona_hakab_model/affinity_matrix.py

```diff
@@ -196,27 +168,23 @@ def clustered_matrix_generation(
 
         # manually generate the first m + 1 connections
         for i in range(m + 1):
-            other_nodes = nodes[0 : m + 1]
+            other_nodes = nodes[0: m + 1]
             other_nodes.pop(i)
             nodes[i].add_connections(other_nodes)
             inserted_nodes.append(nodes[i])
 
-            # add the newly made connections to the connections list
-            connections[0, connections_cnt : connections_cnt + len(other_nodes)] = nodes[i].index
-            connections[1, connections_cnt : connections_cnt + len(other_nodes)] = np.fromiter(
-                [node.index for node in other_nodes], int
-            )
-            connections_cnt += len(other_nodes)
+            # add the newly made connections to the connections list. note that this is one directional,
+            # but the other direction will be added when adding the other's connections
+            connections[nodes[i].index].extend([node.index for node in other_nodes])
 
         # add the rest of the nodes, one at a time
-        for node in nodes[m + 1 :]:
+        for node in nodes[m + 1:]:
```

Comment:
slicing copies the list, use itertools.islice instead

Reply:
done